### ISSUE TITLE
`TC200 - Use 'raise from' to specify cause` is superfluous

### ISSUE THREAD

**Issue by tmarice** [guilatrova/tryceratops#6]
Since Python 3, exception chaining is the default.

https://docs.python.org/3/tutorial/errors.html#exception-chaining

```
Exception chaining happens automatically when an exception is raised inside an except or finally section.
```

**Comment by guilatrova** [comment]
That's very neat! I'm having trouble testing and checking it though. Can you help me?

Python version: 3.9.5
I created the following snippet:

```py
class CustomException(Exception):
    pass

def raiseme():
    raise Exception


try:
    raiseme()
except Exception as ex:
    raise CustomException from ex
```

### Using explicit `from`

The message is clearly: **The above exception was the direct cause of the following exception:**

```
Traceback (most recent call last):
  File "/.../chain.py", line 9, in <module>
    raiseme()
  File "/.../chain.py", line 5, in raiseme
    raise Exception
Exception

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/.../chain.py", line 11, in <module>
    raise CustomException from ex
__main__.CustomException
```

### Bare raise

The message becomes: **During handling of the above exception, another exception occurred:** 

```
Traceback (most recent call last):
  File "/.../chain.py", line 9, in <module>
    raiseme()
  File "/.../chain.py", line 5, in raiseme
    raise Exception
Exception

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/.../chain.py", line 11, in <module>
    raise CustomException
__main__.CustomException
```

### Conclusion

So I wonder whether it's Python 3.9.6 specifically or if I did something wrong. Can you help me test and suggest how we can proceed @tmarice ?

**Comment by tmarice** [comment]
I just tested this on Python 3.6.8, and it displays the same behavior.
It seems that the message is different, but both stack traces are still printed out.

Perhaps there's a slight semantic difference between these two messages -- the former signals the re-raising, while the latter could be caused by an unexpected error in the error handling code.

**Comment by guilatrova** [comment]
After giving some thought to it, I believe we should follow the zen of python:

> Explicit is better than implicit.
[Reference](https://www.python.org/dev/peps/pep-0020/)

This means that I prefer to keep this violation (enforcing user to type `from e`).

Maybe I will change my mind later on, but for now, if you still don't want this rule on your project. I recommend you to ignore this with a [pyproject config](https://github.com/guilatrova/tryceratops#configuration):

```toml
ignore = ["TC200"]
```

**Comment by RojerGS** [comment]
As far as my understanding goes, there _is_ a semantic difference in the two.

Using `raise from` tells Python that we tried handling some exception `Y`, but for some reason we failed/couldn't and then gave up trying to fix it for some reason `X`, and so we raise `X` from trying to handle `Y`.

The other type of error, that shows the message "during handling of the above exception, another exception occurred", just seems like the message that would be displayed after an accident and not like something you'd raise on purpose.

Just leaving my thoughts here for future reference. Let me know if I got something wrong!
